### PR TITLE
Fix Operation in Wrong Order error for Krylov approximation

### PR DESCRIPTION
**Pull request by dburkhardt**
It seems there's an incompatibility with the current version of SLEPC when using a Krylov approximation for getting the sorted schur decomposition.

See recommendation from the SLEPC devs here:
https://gitlab.com/slepc/slepc4py/-/issues/41

Fixing this function removes the error message.

### CODE REVIEW

**Comment by msmdev** [comment]
@michalk8, could you please take a look on this please, since you are more into the SLEPc dependencies topic, esp. with respect to MacOS etc.?

**Comment by msmdev** [comment]
@dburkhardt, could you please provide info regarding the OS and versions of SLEPc, CellRank and pyGPCCA you are using?

**Comment by dburkhardt** [comment]
I'm using Debian GNU/Linux 11 (bullseye) and SLEPC version 3.18 and pyGPCCA version 1.0.3.

CellRank isn't needed to reproduce the issue. It sounds like this is just a new requirement from SLEPC version 3.18 which was released on 2022-10-01. https://gitlab.com/slepc/slepc/-/blob/main/CHANGELOG.md#318-2022-10-01

**Review by michalk8 — CHANGES_REQUESTED** [review]
Thanks @dburkhardt ! Could you please modify the following files and fix the linting please?
- https://github.com/msmdev/pyGPCCA/blob/main/setup.py#L39-L42
- https://github.com/msmdev/pyGPCCA/blob/main/.github/workflows/ci.yml#L65-L67

File: pygpcca/__init__.py [hunk 1]
```diff
@@ -2,5 +2,5 @@
 from pygpcca._gpcca import GPCCA, gpcca_coarsegrain
 
 __author__ = __maintainer__ = "Bernhard Reuter"
-__version__ = "1.0.3"
+__version__ = "1.0.4"
```
Comment (msmdev) [inline pygpcca/__init__.py:5]: @dburkhardt, please don't bump the pyGPCCA version. I will do this separately after adding some other changes.

File: setup.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
     from pygpcca import __email__, __author__, __version__, __maintainer__
 except ImportError:
     __author__ = __maintainer__ = "Bernhard Reuter"
-    __version__ = "1.0.3"
+    __version__ = "1.0.4"
```
Comment (msmdev) [inline setup.py:9]: @dburkhardt, no bumping please.

**Comment by dburkhardt** [comment]
@michalk8 , I edited the request files. I assume you're okay with pinning to the most recent version of SLEPC/PETSC? It seems there is always going to be an incompatibility with versions <3.18 and versions ≥3.18.

**Review by msmdev — CHANGES_REQUESTED** [review]
Please don't bump pyGPCCA version.

**Comment by msmdev** [comment]
@dburkhardt @michalk8 `ubuntu-latest, 3.8, true` still fails with
```
./.scripts/ci/install_dependencies.sh: line 8: pushd: petsc-3.18: No such file or directory
Error: Process completed with exit code 1.
```

**Comment by msmdev** [comment]
@dburkhardt, linting still fails. You should use the `pre-commit` hook to solve this issue.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -63,8 +63,8 @@ jobs:
                     use_slepc: false
         env:
             # make sure to pin this in tox.ini as well
-            PC_VERSION: 3.15.4  # PETSc version
-            SC_VERSION: 3.15.2  # SLEPc version
+            PC_VERSION: 3.18  # PETSc version
```
Comment (michalk8) [inline .github/workflows/ci.yml:66]: This should also include the minor version as `3.18.0`, that's the reason why the CI script fails.

**Comment by dburkhardt** [comment]
I see got it! I think I've made the requested edits correctly. Thanks for the help.

**Comment by msmdev** [comment]
> I see got it! I think I've made the requested edits correctly. Thanks for the help.

@dburkhardt, all tests pass, thus I will merge the PR.
Thank you very much for bringing the issue up and solving it!

**Review by msmdev — APPROVED** [review]

**Comment by msmdev** [comment]
@michalk8, do you approve? If so, please approve the changes to allow the merge, thx!

**Review by michalk8 — APPROVED** [review]

**Comment by michalk8** [comment]
LGTM, thanks @dburkhardt !

**Pull request merged**